bokehjs/test/models/tickers/log_ticker.coffee

```diff
@@ -0,0 +1,50 @@
+{expect} = require "chai"
+utils = require "../../utils"
+
+{LogTicker} = utils.require("models/tickers/log_ticker")
+
+describe "LogTicker Model", ->
+
+  describe "LogTicker get_ticks_no_defaults method", ->
+
+    # short range (<2 base orders)
+
+    it "should have three major ticks and zero minor ticks for long range", ->
+      ticker = new LogTicker({desired_num_ticks: 3, num_minor_ticks: 0})
+      ticks = ticker.get_ticks_no_defaults(1, 999, null, 3)
+      expect(ticks.major).to.be.deep.equal([ 1, 10, 100 ])
+      expect(ticks.minor).to.be.deep.equal([ ])
+
+    it "should have three major ticks and five minor ticks for long range", ->
+      ticker = new LogTicker({desired_num_ticks: 3, num_minor_ticks: 1})
+      ticks = ticker.get_ticks_no_defaults(1, 999, null, 3)
+      expect(ticks.major).to.be.deep.equal([ 1, 10, 100 ])
+      expect(ticks.minor).to.be.deep.equal([ 0.1, 1, 10, 100, 1000 ])
+
+    it "should do stuff on short scale", ->
+      ticker = new LogTicker({desired_num_ticks: 3, num_minor_ticks: 2})
+      ticks = ticker.get_ticks_no_defaults(1, 999, null, 3)
+      expect(ticks.major).to.be.deep.equal([ 1, 10, 100 ])
+      # 0.2 is weird, should be 0.5 i think
+      expect(ticks.minor).to.be.deep.equal([ 0.2, 0.1, 1, 5, 10, 50, 100, 500, 1000 ])
+
+    # long range (>=2 base orders)
+
+    it "should have four major ticks and zero minor ticks for long range", ->
+      ticker = new LogTicker({num_minor_ticks: 0})
+      ticks = ticker.get_ticks_no_defaults(1, 1000, null, 4)
+      expect(ticks.major).to.be.deep.equal([ 1e0, 1e1, 1e2, 1e3 ])
+      expect(ticks.minor).to.be.deep.equal([ ])
+
+    it "should have four major ticks and six minor ticks for long range", ->
+      ticker = new LogTicker({num_minor_ticks: 1})
+      ticks = ticker.get_ticks_no_defaults(1, 1000, null, 4)
+      expect(ticks.major).to.be.deep.equal([ 1e0, 1e1, 1e2, 1e3 ])
+      expect(ticks.minor).to.be.deep.equal([ 1e-1, 1e0, 1e1, 1e2, 1e3, 1e4 ])
+
+    it "should have four major ticks and ten minor ticks for long range", ->
+      ticker = new LogTicker({num_minor_ticks: 2})
+      ticks = ticker.get_ticks_no_defaults(1, 1000, null, 4)
+      expect(ticks.major).to.be.deep.equal([ 1e0, 1e1, 1e2, 1e3 ])
+      # 0.2 is weird, should be 0.5 i think
+      expect(ticks.minor).to.be.deep.equal([ 0.2, 1e-1, 1e0, 5e0, 1e1, 5e1, 1e2, 5e2, 1e3, 5e3, 1e4 ])
```

Comment:
Also for later: we need to figure out a way to not have minor ticks outside bounds when explicit axis bounds for ticks are present.